pennylane/drawer/drawable_layers.py

```diff
@@ -115,41 +147,51 @@ def drawable_layers(ops, wire_map=None):
     transformation.
 
     """
-    # pylint:disable=too-many-nested-blocks,too-many-branches,too-many-statements
 
-    if not wire_map:
-        wire_map = default_wire_map(ops)
-    # initialize
+    wire_map = wire_map or default_wire_map(operations)
+    bit_map = bit_map or {}
+
+    # initialize for operation layers
     max_layer = 0
     occupied_wires_per_layer = [set()]
     ops_in_layer = [[]]
-    measured_wires = {}
-
-    # Collect all mid-circuit measurements used for classical conditioning
-    cond_measurements = set()
-    for op in ops:
-        if isinstance(op, Conditional):
-            cond_measurements.update(op.meas_val.measurements)
+    used_cwires_per_layer = [set()]
 
     # loop over operations
-    for op in ops:
+    for op in operations:
         if isinstance(op, MidMeasureMP):
             if len(op.wires) > 1:
                 raise ValueError("Cannot draw mid-circuit measurements with more than one wire.")
 
-            measured_wires[op.id] = wire_map[op.wires[0]]
-
-        op_occupied_wires = _get_op_occupied_wires(op, wire_map, cond_measurements)
-        op_layer = _recursive_find_layer(max_layer, op_occupied_wires, occupied_wires_per_layer)
+        if not (mv := getattr(op, "mv", None)):
+            # Only terminal measurements that collect mid-circuit measurement statistics have
+            # op.mv != None
+            op_occupied_wires = _get_op_occupied_wires(op, wire_map, bit_map)
+            op_layer = _recursive_find_layer(max_layer, op_occupied_wires, occupied_wires_per_layer)
+            op_occupied_cwires = set()
+
+        else:
+            op_occupied_wires = set()
+            mapped_cwires = (
+                [bit_map[m.measurements[0]] for m in mv]
+                if isinstance(mv, list)
+                else [bit_map[m] for m in mv.measurements]
+            )
+            op_occupied_cwires = set(range(min(mapped_cwires), max(mapped_cwires) + 1))
+            op_layer = _recursive_find_mcm_stats_layer(
+                max_layer, op_occupied_cwires, used_cwires_per_layer, bit_map
+            )
```

Comment:
Can we switch the order of these two branches? It's often easier to read if we can control on the positive (`condition`) then the negative (`not condition`) . The `not` gives you one more piece of information to carry around in your active memory.

Also, even though `getattr(op, "mv", None)` does indeed get the job done, it might not be as readable to someone less familiar with mid circuit measurements.  Maybe instead `isinstance(op, MeasurementProcess) and op.mv`?